docs/source/index.rst

```diff
@@ -0,0 +1,275 @@
+        return "Hello world!"
+
+    if __name__ == "__main__":
+        app.run()
+
+Now we need to set up the configuration: namespace of plugins we will be using with the application and plugins we want to use.
+Let's use the ``example_plugin`` we created before. For example::
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+Next we need to create an instance of PluginEngine for our app and load the plugins::
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+
+Then we can access loaded plugins by calling :func:`get_active_plugins` method, it will return a dictionary.
+If we want to check if all of the plugins were loaded correclty we can also call :func:`get_failed_plugins` method that will return a dictionary with plugins that failed to load::
+
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+
+Example application functionality
+---------------------------------
+
+To understand the possibilities of Flask-PluginEngine we will create a Jinja template, where we will list all the plugins we are using.
+Let's create two templates, the base one (base.html), for example::
+
+    <!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title>Title</title>
+    </head>
+    <body>
+        {% block content %}
+        {% endblock %}
+    </body>
+    </html>
+
+And index.html, inheriting from the base one, for example::
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        {% for plugin in plugins %}
+            <div>{{ plugin.title }} - {{ plugin.description }}</div>
+        {% endfor %}
+    {% endblock %}
+
+And let's pass the plugins, when rendering the template in our application::
+
+    @app.route("/")
+    def hello():
+    return render_template('index.html', plugins=active_plugins)
+
+Now what we also should do is register the blueprints of all our plugins, for instance::
+
+    for plugin in active_plugins:
+    with plugin.plugin_context():
+        app.register_blueprint(plugin.get_blueprint())
+
+So our application code looks like this::
+
+    from flask import render_template
+    from flask_pluginengine import PluginFlask, PluginEngine
+
+    app = PluginFlask(__name__)
+
+    app.config['PLUGINENGINE_NAMESPACE'] = 'example'
+    app.config['PLUGINENGINE_PLUGINS'] = ['example_plugin']
+
+    plugin_engine = PluginEngine(app=app)
+    plugin_engine.load_plugins(app)
+    active_plugins = plugin_engine.get_active_plugins(app=app).values()
+
+    for plugin in active_plugins:
+        with plugin.plugin_context():
+            app.register_blueprint(plugin.get_blueprint())
+
+
+    @app.route("/")
+    def hello():
+        return render_template('index.html', plugins=active_plugins)
+
+
+    if __name__ == "__main__":
+        app.run()
+
+Now when we will go to the index page of our application we will be able to access plugin's template.
```

Comment:
"Now, when we ~will~ go to the index page of our application we ~will be able to~ **can** access **the** plugin's template."